src/app/hooks/useGa.ts

```diff
@@ -20,16 +16,21 @@ interface InitGaProps {
 
 const initGa4 = ({ ga4MeasurementId, debugMode }: InitGaProps) => {
   // Never run in tests:
-  /* c8 ignore next 3 */
+  /* c8 ignore next 8 */
   if (debugMode) {
-    console.info("Initialize GA4");
+    console.info("Initialize GA4 debug mode");
+  }
+
+  if (typeof navigator !== "undefined" && navigator.doNotTrack === "1") {
+    console.info("doNotTrack enabled, not initializing GA4");
+    return;
   }
 
   // GA4 setup
   window.dataLayer = window.dataLayer || [];
   if (!window.gtag) {
     window.gtag = function (...args: unknown[]) {
-      window.dataLayer.push(args);
+      window.dataLayer?.push(args);
```

Comment:
Why the `?.` here if we define it above on L30?

Reply:
I'm not sure, the typescript compiler claims that it might be empty. I intend to replace most of the code in here per the TODO though, I just wanted to try to make the current fix as minimal as possible.